### PR TITLE
✨ Add extract sas helper function

### PR DESCRIPTION
**Pull request by s-weigand**
Just a little helper function to easily extract a single SAS which can then be used as clp guidance.
Requested by @ism200 

### Change summary

- [👌 Allow passing of prepare argument to load_dataset ascii](https://github.com/glotaran/pyglotaran/commit/dcd3fabf40609f536deb48eee1c4e75d11bf31fb)
- [✨ Added 'extract_sas' helper function](https://github.com/glotaran/pyglotaran/commit/3de4c4b899f5a036b882fac31dda581f547474d4)

### Checklist

- [x] ✔️ Passing the tests (mandatory for all PR's)
- [x] 🚧 Added changes to changelog (mandatory for all PR's)
- [x] 🧪 Adds new tests for the feature (mandatory for ✨ feature and 🩹 bug fix PR's)

### CODE REVIEW

**Comment by s-weigand** [comment]
For usage instructions [see the docs](https://pyglotaran--1015.org.readthedocs.build/en/1015/api/glotaran/utils/io/glotaran.utils.io.extract_sas.html#glotaran.utils.io.extract_sas)

File: glotaran/utils/io.py [hunk 1]
```diff
@@ -252,3 +253,95 @@ def safe_dataframe_replace(
...
+    """Extract SAS data from a result.
+
+    This is a helper function to easily extract clp guidance spectra.
+
+    Parameters
+    ----------
+    result: Result | xr.Dataset
+        Optimization ``Result`` instance or result dataset.
+    species: str | None
+        Name op the species to extract the SAS for. Defaults to None
```
Comment (jsnel) [inline glotaran/utils/io.py:270]: "Name op" -> "Name of"

File: glotaran/utils/io.py [hunk 2]
```diff
@@ -252,3 +253,95 @@ def safe_dataframe_replace(
...
+
+    .. code-block:: python
+
+        from glotaran.io import load_result
+        from glotaran.io import save_dataset
+        from glotaran.utils.io import extract_sas
+
+        result_dataset = load_result("result_dataset_1.nc")
+        sas = extract_sas(result_dataset, "species_1")
+        save_dataset("sas__result_dataset_1__species_1.ascii")
```
Comment (jsnel) [inline glotaran/utils/io.py:313]: see earlier comment `save_dataset(sas, 'filename.ascii)`

**Review by jsnel — CHANGES_REQUESTED** [review]
Some minor issues with documentation to fix, otherwise seems to work fine. 
The error messages are a good addition and useful, e.g.:

> ValueError: The result doesn't contain a species with name 'species_1'.
> Valid values are: ['s1', 's2', 's3', 's4']

File: glotaran/utils/io.py (hunk 2 above)
Comment (s-weigand) [reply glotaran/utils/io.py]: Fixed in [90146e5](https://github.com/glotaran/pyglotaran/pull/1015/commits/90146e5d6c90e55e52795ed357f937e2ac1df19f)

File: glotaran/utils/io.py [hunk 3]
```diff
@@ -252,3 +253,95 @@ def safe_dataframe_replace(
...
+    --------
+    Extracting the SAS from an optimization result object.
+
+    .. code-block:: python
+
+        from glotaran.utils.io import extract_sas
+        from glotaran.io import save_dataset
+
+        sas = extract_sas(result, "species_1","dataset_1")
+        save_dataset("sas__result_dataset_1__species_1.ascii")
```
[the remark this answers is not in the document]
Comment (s-weigand) [reply glotaran/utils/io.py]: Fixed in [90146e5](https://github.com/glotaran/pyglotaran/pull/1015/commits/90146e5d6c90e55e52795ed357f937e2ac1df19f)

File: glotaran/utils/io.py (hunk 1 above)
Comment (s-weigand) [reply glotaran/utils/io.py]: Fixed in [90146e5](https://github.com/glotaran/pyglotaran/pull/1015/commits/90146e5d6c90e55e52795ed357f937e2ac1df19f)

**Comment by s-weigand** [comment]
After some discussion with @joernweissenborn, we decided to drop this PR in favor of a more generic and reusable implementation as method `create_clp_guidance_dataset` on the `Result` object utilizing the `clp` data variable of the result datasets.

**Pull request closed without merging**